### ISSUE TITLE
Grafana update v8.1.7 to v8.2.5 crashes dashboads "Dashboard init failed"

### ISSUE THREAD

**Issue by stvolinko** [grafana/grafana#42643]
**What happened**:
After update Grafana to v8.2.5 list of dashboards with panel "Zabbix problems"  fails with "Dashboard init failed" error.

Screenshot pre update:
![Screenshot_1](https://user-images.githubusercontent.com/49945693/144419402-805167bb-566e-4b6e-bd6d-b28075b0f098.jpg)
Screenshop post update:
![Screenshot_2](https://user-images.githubusercontent.com/49945693/144419448-08cff97e-d518-494b-9c50-44450ceca2fd.png)

With browser dev console i can see error in initDashboard.ts:137:
`TypeError: object is not iterable (cannot read property Symbol(Symbol.iterator))
    at C.migrateCloudWatchQueries (DashboardMigrator.ts:718)
    at C.<anonymous> (DashboardMigrator.ts:691)
    at C.updateSchema (DashboardMigrator.ts:704)
    at Z.updateSchema (DashboardModel.ts:1049)
    at new Z (DashboardModel.ts:163)
    at initDashboard.ts:134
(anonymous) @ initDashboard.ts:137`

There is no errors in log this time(level=debug).

**How to reproduce it (as minimally and precisely as possible)**:
Im using to test this error this docker-compose [set-up](https://github.com/guedim/grafana-postgres/blob/master/docker-compose.yml)
- Starting with image: grafana/grafana:8.1.7
-  import dashboard:  [Triggers-1638447699471.zip](https://github.com/grafana/grafana/files/7641792/Triggers-1638447699471.zip)
- stop grafana container and update it to v8.2.7
- dashboard broken.

**Environment**:
- Grafana version: 8.1.7\8.2.5
- Data source type & version: alexanderzobnin-zabbix-app (4.2.4\4.1.5)
- OS Grafana is installed on: docker for test 
- User OS & Browser: Windows Chrome/FireFox
- Grafana plugins :alexanderzobnin-zabbix-app (4.2.4\4.1.5)

**Comment by dprokop** [comment]
@stvolinko I've tested the provided dashboard against 8.1.7 (it worked) and then upgraded 8.2.5 and I couldn't see any problem with the dashboard provided as an attachment.

Just to make sure - you don't have any cloudwatch queries in the original dashboard?

cc @sunker - do you have any ideas or heard about similar issue? Seems like the cloudwatch queries migrator fails.

**Comment by sunker** [comment]
Yeah but not really sure what's happening here. @torkelo do you know? 
https://github.com/grafana/grafana/blob/b57a137acd75de86455dbbec89c00ae86a8bde08/public/app/features/dashboard/state/DashboardMigrator.ts#L718

**Comment by stvolinko** [comment]
@dprokop There is no cloudwatch queries 100%. Its zabbix problem pannel with query trhrough zabbix plugin
![изображение](https://user-images.githubusercontent.com/49945693/144560596-5d8ab757-f018-477a-929f-7d2378fe6154.png)

**Comment by sunker** [comment]
Thanks for additional info @stvolinko. 

I tried uploading the attached dashboard in 8.2.5, but was not able to reproduce the problem unfortunately. Judging by the error message, it seems like the targets array in the panel is an object an not an array, but that doesn't seem to be the case when I inspect the dashboard json. 

@alexanderzobnin do you have any ideas?

**Comment by stvolinko** [comment]
Realized the thing that i missed in test case:
Grafana should be connected to zabbix server datasource. 
I will try to make another compose file with all prerequisites.

**Comment by alecxvs** [comment]
For the workaround, I ended up combing through the whole panel JSON and making sure it was updated to the "current" schema as defined by the newly created panel. A bit tedious, but if I just updated with the fixed targets, it would wipe out the panel datasource and targets. There's probably specific fields I needed to update to prevent that but I'm not sure what those are yet.

**Comment by gonzalo-** [comment]
Any fix for this? I am having the very same issue.
